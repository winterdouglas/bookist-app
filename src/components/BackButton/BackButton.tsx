import { Icon } from "@/components/Icon";
import { useTheme } from "@/hooks/useTheme";
import { spacing } from "@/theme";
import { useNavigation } from "@react-navigation/native";
import { StyleSheet, TouchableOpacity } from "react-native";

export const BackButton = () => {
  const navigation = useNavigation();
  const { colors } = useTheme();

  return (
    <TouchableOpacity
      onPress={navigation.goBack}
      style={[styles.container, { backgroundColor: colors.text }]}>
      <Icon
        size={spacing.extraLarge}
        name="chevron-back"
        style={[styles.icon, { color: colors.background }]}
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: spacing.huge / 2,
    width: spacing.huge,
    height: spacing.huge,
    opacity: 0.8,
    justifyContent: "center",
    alignItems: "center",
  },
  icon: {
    marginRight: spacing.tiny,
  },
});
