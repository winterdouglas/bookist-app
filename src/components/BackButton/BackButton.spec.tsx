import renderer from "react-test-renderer";
import { BackButton } from "./BackButton";
import { ThemeProvider } from "@/contexts";
import { NavigationContainer } from "@react-navigation/native";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <ThemeProvider>
          <BackButton />
        </ThemeProvider>
      </NavigationContainer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
