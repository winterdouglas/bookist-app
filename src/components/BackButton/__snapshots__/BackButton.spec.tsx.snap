// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(28, 28, 30)",
      "borderRadius": 16,
      "height": 32,
      "justifyContent": "center",
      "opacity": 0.8,
      "width": 32,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        [
          {
            "marginRight": 4,
          },
          {
            "color": "rgb(242, 242, 242)",
          },
        ],
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î¬ª
  </Text>
</View>
`;
